csharp/src/Glacier2/SessionHelper.cs

```diff
@@ -12,130 +12,69 @@
-        /// <param name="finderStr">The stringified Ice.RouterFinder proxy.</param>
-        /// <param name="useCallbacks">True if the session should create an object adapter for receiving callbacks.</param>
-        internal SessionHelper(ISessionCallback callback,
-            string finderStr,
-            bool useCallbacks,
-            Dictionary<string, string> properties,
-            ILogger? logger = null,
-            ICommunicatorObserver? observer = null,
-            X509Certificate2Collection? certificates = null,
-            X509Certificate2Collection? caCertificates = null,
-            ICertificateVerifier? certificateVerifier = null,
-            IPasswordCallback? passwordCallback = null)
+        /// <summary>Gets the session's communicator object.</summary>
+        public Communicator? Communicator
         {
-            _callback = callback;
-            _finderStr = finderStr;
-            _useCallbacks = useCallbacks;
-            _properties = properties;
-            _logger = logger;
-            _observer = observer;
-            _certificates = certificates;
-            _caCertificates = caCertificates;
-            _certificateVerifier = certificateVerifier;
-            _passwordCallback = passwordCallback;
-        }
-
-        /// <summary>
-        /// Destroys the Glacier2 session.
-        ///
-        /// Once the session has been destroyed, SessionCallback.disconnected is
-        /// called on the associated callback object.
-        /// </summary>
-        public void
-        Destroy()
-        {
-            lock (_mutex)
+            get
             {
-                if (_destroy)
-                {
-                    return;
-                }
-                _destroy = true;
-                if (!_connected)
+                lock (_mutex)
                 {
-                    //
-                    // In this case a connecting session is being destroyed.
-                    // We destroy the communicator to trigger the immediate
-                    // failure of the connection establishment.
-                    //
-                    var t1 = new Thread(new ThreadStart(DestroyCommunicator));
-                    t1.Start();
-                    return;
+                    return _communicator;
                 }
-                _session = null;
-                _connected = false;
-
-                //
-                // Run destroyInternal in a thread because it makes remote invocations.
-                //
-                var t2 = new Thread(new ThreadStart(DestroyInternal));
-                t2.Start();
             }
         }
 
-        /// <summary>
-        /// Returns the session's communicator object.
-        /// </summary>
-        /// <returns>The communicator.</returns>
-        public Communicator?
-        Communicator()
+        /// <summary>Gets whether there is a need an active session</summary>
```

Comment:
Feel like there's a word or two missing in this comment.

Reply:
fixed